### ISSUE TITLE
False positive no-name-in-module with gspread_formatting

### ISSUE THREAD

**Issue by sam-s** [PyCQA/pylint#3835]
### Steps to reproduce
1. Run `pip install gspread_formatting`
2. Put `from gspread_formatting import set_frozen, format_cell_ranges` in file `z.py`
3. Run `pylint z.py`

### Current behavior

```
z.py:1: [E0611(no-name-in-module), ] No name 'set_frozen' in module 'gspread_formatting'
z.py:1: [E0611(no-name-in-module), ] No name 'format_cell_ranges' in module 'gspread_formatting'
```
(plus 3 other - valid - errors)

### Expected behavior

no `no-name-in-module` errors

### pylint --version output
```
pylint 2.6.0
astroid 2.4.2
Python 3.8.5 (default, Jul 21 2020, 10:48:26) 
[Clang 11.0.3 (clang-1103.0.32.62)]
```

**Comment by hippo91** [comment]
@sam-s thanks for the report. I can reproduce it.

**Comment by PCManticore** [comment]
Thanks for the report. That module is added dynamically and exposed through `gpsread_formatting.batch` over here https://github.com/robin900/gspread-formatting/blob/a77ffc9f17e18e39b94303910c9412e791d1f8f9/gspread_formatting/batch.py#L44

`pylint` is not capable of understanding dynamic code like this. There are multiple flags that you can use to tell `pylint` to either ignore the module or to *patch* the module at runtime with the missing attributes.

**Comment by sam-s** [comment]
> There are multiple flags that you can use to tell `pylint` ... to _patch_ the module at runtime with the missing attributes.

Could you please point me to the doc that would explain that?
Thank you!

**Comment by hippo91** [comment]
@sam-s you can have a look to the `extension-pkg-whitelist` and also `ignored-modules` parameters in your rcfile.

**Comment by sam-s** [comment]
> @sam-s you can have a look to the `extension-pkg-whitelist` and also `ignored-modules` parameters in your rcfile.

this tells me how to make pylint _ignore_ a package. 
how do I tell pylint _how_ to check a package?

**Comment by hippo91** [comment]
@sam-s the `extension-pkg-whitelist` is what you are looking for [http://pylint.pycqa.org/en/latest/technical_reference/c_extensions.html](http://pylint.pycqa.org/en/latest/technical_reference/c_extensions.html).
